### DISCUSSION TITLE
PocketHost outage Jan 19, 2023

### DISCUSSION THREAD

**Discussion by mtaku3** [pockethost/pockethost#169] [Q&A] [answered]
Website seems to be really unstable, slow to load and can't log in. Also, my instance doesn't respond.
Any information?

**Comment by benallfree** [comment]
@mtaku3 Thanks for the report. I think #168 may still be in progress, we might not have found the root cause. Something started happening recently and I haven't been able to pinpoint it yet.

As you can see, the CPU is pegged at 100% which generally indicates a runaway process of some type.

<img width="1033" alt="image" src="https://user-images.githubusercontent.com/1068356/213610203-92df74ff-bb01-4303-87ef-b43857a2567e.png">

I'll investigate a bit if I can, and reboot the server in the meantime.

**Comment by benallfree** [comment]
The issue seems to be that the daemon process (node) is restarting. When it does, `pm2` is not killing the child processes that it launches. I found 189 restarts of the daemon. It really should never be crashing in the first place, but in theory, `pm2` is supposed to gracefully restart on a crash. Except, for some reason the old `pocketbase` child processes are not getting cleaned up. Eventually the server runs out of resources. That's what is happening.


<img width="941" alt="image" src="https://user-images.githubusercontent.com/1068356/213610939-76bb4331-d2f0-4fee-9fc8-ef9e4cfa461c.png">

I'll continue investigating, but the service should be back up now.

**Comment by pyloverm** [comment]
The issue might be happening again.

**Reply by benallfree** [reply]
Appears to be.

**Reply by benallfree** [reply]
Ok we're up again, but the main cause is still under investigation. I expect this will keep happening until I isolate this issue. It is something that only happens at scale, so I think we'll just need some time to isolate it.

**Comment by RA9** [comment]
It's happening currently

**Reply by RA9** [reply]
I guess I will have to wait for  now

**Comment by TechStudent10** [comment]
Dang it it's happening to me. I've been too busy setting up external services to notice the outage lol. Hope it gets fixed.

**Comment by benallfree** [comment]
Fixed.

Apologies everyone, it's going to keep happening until I can figure out why PocketBase child processes aren't exiting when node exits.

**Reply by TechStudent10** [reply]
Thank you! GL on figuring out the issue. This is very interesting and I personally don't understand why it's happening.

**Comment by benallfree** [comment, marked as answer]
Update for everyone - I've been slowly tracing log files and cleaning up possible conditions where errors might occur that cause the node process to exit and restart. This still doesn't solve the mystery of why the child processes aren't being cleaned up properly, but it does reduce the chances of it causing a server overload.

More info as I gather it :)
